### PR TITLE
steval-eth001v1: add FOC examples

### PR DESCRIPTION
**Pull request by raiden00pl**
## Summary
steval-eth001v1: add FOC examples
depends on https://github.com/apache/incubator-nuttx/pull/6811
## Impact
None

## Testing
examples/foc (float and fixed16 options)

### CODE REVIEW

**Comment by xiaoxiang781216** [comment]
@raiden00pl some build error need to fix before merge:
https://github.com/apache/incubator-nuttx/runs/7839035478?check_suite_focus=true

File: boards/arm/stm32f7/steval-eth001v1/include/board.h [hunk 1]
```diff
@@ -35,7 +35,7 @@
  * Pre-processor Definitions
  ****************************************************************************/
 
-/* Clocking */
+/* Clocking *****************************************************************/
```
Comment (pkarashchenko) [inline boards/arm/stm32f7/steval-eth001v1/include/board.h:38]: IMO this is not needed

**Review by pkarashchenko — APPROVED** [review]

**Review by xiaoxiang781216 — APPROVED** [review]

File: boards/arm/stm32f7/steval-eth001v1/include/board.h (hunk 1 above)
Comment (xiaoxiang781216) [reply boards/arm/stm32f7/steval-eth001v1/include/board.h:38]: But it's common to use the style too.

Comment (pkarashchenko) [reply boards/arm/stm32f7/steval-eth001v1/include/board.h:38]: Sure. No objections!

**Pull request merged**